src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -1178,31 +1226,55 @@ private void setupRootDevice(List<BlockDeviceMapping> deviceMappings) {
         return newDeviceMapping;
     }
 
-    private void setupEphemeralDeviceMapping(List<BlockDeviceMapping> deviceMappings) {
+    private void setupEphemeralDeviceMapping(Image image, List<BlockDeviceMapping> deviceMappings) {
         // Don't wipe out pre-existing mappings
-        deviceMappings.addAll(getNewEphemeralDeviceMapping());
+        deviceMappings.addAll(getNewEphemeralDeviceMapping(image));
     }
 
-    private List<BlockDeviceMapping> getAmiBlockDeviceMappings() {
-
-        /*
-         * AmazonEC2#describeImageAttribute does not work due to a bug
-         * https://forums.aws.amazon.com/message.jspa?messageID=231972
-         */
-        return getImage().getBlockDeviceMappings();
+    @NonNull
+    private static List<String> makeImageAttributeList(@CheckForNull String attr) {
+        return Stream.of(Util.tokenize(Util.fixNull(attr)))
+            .collect(Collectors.toList());
     }
 
-    private Image getImage() {
-        DescribeImagesRequest request = new DescribeImagesRequest().withImageIds(ami);
-        for (final Image image : getParent().connect().describeImages(request).getImages()) {
+    @NonNull
+    private DescribeImagesRequest makeDescribeImagesRequest() {
+        List<String> imageIds = Util.fixEmpty(ami) == null ?
```

Comment:
Sorry if my question is a nonsense, but any reason to use `Util.fixEmpty` instead of `fixEmptyAndTrim`? if ami is "   ", that would result on a List with the unique value "   "
```suggestion
        List<String> imageIds = fixEmptyAndTrim(ami) == null ?
```

Reply:
I think because the plugin was already taking whatever was in `ami` I had decided that the only thing I wanted to detect was the empty string that would result from the form being filled in. But I agree that trimming would be more robust.